staging/src/k8s.io/externaljwt/pkg/plugin/keycache.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"golang.org/x/sync/singleflight"
+	"k8s.io/externaljwt"
+	"k8s.io/klog/v2"
+
+	externaljwtv1alpha1 "k8s.io/externaljwt/apis/v1alpha1"
+	externaljwtmetrics "k8s.io/externaljwt/pkg/metrics"
+)
+
+const fallbackRefreshDuration = 10 * time.Second
+
+type keyCache struct {
+	client externaljwtv1alpha1.ExternalJWTSignerClient
+
+	syncGroup     singleflight.Group
+	listenersLock sync.Mutex
+	listeners     []externaljwt.Listener
+
+	// last successful refresh hint expires at this time.
+	nextRefreshHint atomic.Pointer[time.Time]
+	publicKeys      atomic.Pointer[VerificationKeys]
```

Comment:
I think these should just be a single atomic and `VerificationKeys` should just store the refresh time as a field.

Reply:
Done